lib/sinatra/base.rb

```diff
@@ -348,7 +348,7 @@ def attachment(filename = nil, disposition = :attachment)
       response['Content-Disposition'] = disposition.to_s
       if filename
         params = '; filename="%s"' % File.basename(filename)
-        response['Content-Disposition'] << params
+        response['Content-Disposition'] += params
```

Comment:
@aselder Thanks for your investigation!

I will consider your patch, and this one but unfortunately I don't want to merge all of these comments in the test files.